cmd/nanotube/listen.go

```diff
@@ -124,25 +124,28 @@ loop:
 	close(queue)
 }
 
-func listenUDP(conn net.Conn, queue chan string, stop <-chan struct{},
-	UDPbBufSize int, lg *zap.Logger, ms *metrics.Prom) {
+func listenUDP(conn net.Conn, queue chan string, stop <-chan struct{}, lg *zap.Logger, ms *metrics.Prom) {
 	defer func() {
 		cerr := conn.Close()
 		if cerr != nil {
 			lg.Error("closing the incoming connection", zap.Error(cerr))
 		}
 	}()
 
-	r := bufio.NewReaderSize(conn, UDPbBufSize)
+	buf := make([]byte, 64*1024) // 64k is the max UDP datagram size
 	for {
-		line, err := r.ReadBytes('\n')
-		if err != nil && err != io.EOF {
-			lg.Error("error reading a line from UDP connection", zap.Error(err))
+		nRead, err := conn.Read(buf)
+		if err != nil {
+			lg.Error("error reading UDP datagram", zap.Error(err))
 			continue
 		}
 
+		lines := bytes.Split(buf[:nRead], []byte{'\n'})
+
 		// TODO (grzkv): string -> []bytes
```

Comment:
` 		// TODO (grzkv): string -> []bytes  and pay attention to race conditions on "buf"`

Reply:
;) Yes, yes, the `Split` does not copy.

Reply:
Added.